server/src/printers/TestMakefilesPrinter.cpp

```diff
@@ -62,36 +63,54 @@ namespace printer {
         fs::path sharedMakefilePath = getMakefilePathForShared(generalMakefilePath);
         fs::path objMakefilePath = getMakefilePathForObject(generalMakefilePath);
 
+        generalMakefilePrinter.declareVariable("project_dir_relative_to_makefile",
+                                               fs::relative(projectContext.projectPath, path));
+        // magic spell from https://stackoverflow.com/questions/18136918/how-to-get-current-relative-directory-of-your-makefile
+        generalMakefilePrinter.declareVariable("makefile_dir", "$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))");
+        // TODO: maybe export will work some day?
+        generalMakefilePrinter.declareVariable("project_dir",
+                                               StringUtils::stringFormat("%s/%s", "$(makefile_dir)", "$(project_dir_relative_to_makefile)"));
+        generalMakefilePrinter.declareVariable("build_relative", objMakefilePrinter.buildDirectoryRelative);
+        generalMakefilePrinter.declareVariable("build_dir", StringUtils::stringFormat("%s/%s",
+                                                  objMakefilePrinter.getRelativePath(projectContext.projectPath),
+                                                  objMakefilePrinter.getRelativePath(objMakefilePrinter.buildDirectoryRelative)));
+
```

Comment:
it's all look like copy-paste maybe move to function or some other way not copy it

Reply:
fixed by adding export